### PR TITLE
Return full response from the request rather than throwing an exception

### PR DESCRIPTION
**Pull request by hotmeteor**
Returns `\GuzzleHttp\Psr7\Response` from the webhook request rather than parsing the status code and succeeding silently or throwing an exception.

This functionality allows the app to handle response data itself and creating logging and further action based on statuses or data.

### CODE REVIEW

**Comment by jamesgraham** [comment]
Great addition, thanks

**Comment by hotmeteor** [comment]
Will this ever get merged?

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 build
 composer.phar
 composer.lock
+/.idea
```
Comment (atymic) [inline .gitignore:5]: I don't think this is needed. You can add it to your global gitignore if you use PHPstorm :)

File: composer.json [hunk 2]
```diff
@@ -11,16 +11,16 @@
         }
     ],
     "require": {
-        "php": ">=5.6.4",
+        "php": ">=7.1",
         "guzzlehttp/guzzle": "~6.0",
-        "illuminate/notifications": "5.3.*|5.4.*|5.5.*",
-        "illuminate/support": "5.1.*|5.2.*|5.3.*|5.4.*|5.5.*"
+        "illuminate/notifications": "5.3.*|5.4.*|5.5.*|5.6.*|5.7.*|5.8.*",
```
Comment (atymic) [inline composer.json:16]: Once l5.8 and l6.0 support are a rebase will be needed.

**Comment by atymic** [comment]
This would be a BC break, as code expecting to handle exceptions won't work anymore.

Did you consider changing the thrown exception to have a `response` property, and setting that when the exception is thrown (so you can handle it if required)?

Otherwise, we could also have a config option (throw on request failure).

**Comment by hotmeteor** [comment]
Great feedback, I haven't looked at this in a while. I'll look at making the suggested changes.

File: .gitignore (hunk 1 above)
Comment (hotmeteor) [reply .gitignore:5]: Cool, didn't know about that. Shouldn't have included that either way.

**Comment by atymic** [comment]
No worries, sorry for the delays. I'll try and be responsive to future PRs :)

**Comment by hotmeteor** [comment]
I like the idea of adding it to the exception, but I think returning it in the event it's successful is also useful. For example, we have a system that records the response codes of webhooks so we know if we need to retry or not.

**Comment by atymic** [comment]
> I like the idea of adding it to the exception, but I think returning it in the event it's successful is also useful. For example, we have a system that records the response codes of webhooks so we know if we need to retry or not.

That's fine IMO, as it shouldn't break existing code.

**Comment by hotmeteor** [comment]
Closing this PR in favor a simpler version on a different branch: https://github.com/laravel-notification-channels/webhook/pull/28

**Pull request closed without merging**